### PR TITLE
Enhance secrets documentation

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area documentation
/kind enhancement

**Which issue(s) this PR fixes**:
Fixes a part of #3405

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/usage/secrets_rotation.md [hunk 1]
```diff
@@ -18,3 +23,43 @@ To put it in use, a cloudprovider secret is bound to one more namespaces (and th
...
+**NOTE**
+
+This Kubeconfig contains the highest privileges in the cluster. We strongly discourage distributing or using this Kubeconfig
+on a daily basis. Use dedicated [Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
+[OIDC](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens) or similar alternatives
+to grant role-based and revocable access for a broader audience.
+
+---
+
+*Rotation*: Kubeconfig can be rotated by annotating the shoot resource with `gardener.cloud/operation: rotate-kubeconfig-credentials`.
```
Comment (vpnachev) [inline docs/usage/secrets_rotation.md:45]: Long time ago this kubeconfig was containing a client certificate that cannot be revoked easily. See https://github.com/gardener/gardener/pull/1339 for more details. Should we mention this limitation in the documentation?

File: docs/usage/secrets_rotation.md [hunk 2]
```diff
@@ -18,3 +23,43 @@ To put it in use, a cloudprovider secret is bound to one more namespaces (and th
 1. Update the data keys in the secret.
 2. :warning: Wait until all Shoots using the secret are reconciled before you disable the old secret in your infrastructure account! Otherwise the shoots will no longer function.
 3. After all Shoots using the secret were reconciled you can go ahead and deactivate the old secret in your infrastructure account.
+
+### Gardener generated secrets
+
+####Kubeconfig
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]:
```suggestion
#### Kubeconfig
```

File: docs/usage/secrets_rotation.md [hunk 3]
```diff
@@ -18,3 +23,43 @@ To put it in use, a cloudprovider secret is bound to one more namespaces (and th
...
+
+*Name*: <shoot-name>.kubeconfig
+
+*Description*: Admin Kubeconfig provided by Gardener for the managed shoot cluster.
+
+---
+**NOTE**
+
+This Kubeconfig contains the highest privileges in the cluster. We strongly discourage distributing or using this Kubeconfig
+on a daily basis. Use dedicated [Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]:
```suggestion
This Kubeconfig contains the highest privileges in the cluster. We strongly discourage distributing or using this Kubeconfig. Use dedicated [Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
```

File: docs/usage/secrets_rotation.md [hunk 4]
```diff
@@ -18,3 +23,43 @@ To put it in use, a cloudprovider secret is bound to one more namespaces (and th
...
+[OIDC](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens) or similar alternatives
+to grant role-based and revocable access for a broader audience.
+
+---
+
+*Rotation*: Kubeconfig can be rotated by annotating the shoot resource with `gardener.cloud/operation: rotate-kubeconfig-credentials`.
+The substituted Kubeconfig are provided after the initialized reconciliation was performed. 
+
+####Monitoring
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]:
```suggestion
#### Monitoring
```

File: docs/usage/secrets_rotation.md [hunk 5]
```diff
@@ -18,3 +23,43 @@ To put it in use, a cloudprovider secret is bound to one more namespaces (and th
...
+
+*Name*: <shoot-name>.monitoring
+
+*Description*: Username/password for accessing the user Grafana instance of a shoot cluster.
+
+*Rotation*: Not supported yet.
+
+###SSH-Keypair
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]:
```suggestion
### SSH-Keypair
```

**Comment by timuthy** [comment]
Thanks for your reviews @vpnachev and @rfranzke. PTAL.

**Review by vpnachev — APPROVED** [review]
/lgtm
/milestone v1.32

**Review by rfranzke — APPROVED** [review]

**Pull request merged**